src/features/commentsSlice.ts

```diff
@@ -0,0 +1,102 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Comment, NewCommentData } from '../types/Comment';
+import {
+  createComment,
+  deleteComment,
+  getPostComments,
+} from '../api/comments';
+
+type CommentsState = {
+  comments: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+  visible: boolean;
+  error: string;
+};
+
+const initialState: CommentsState = {
+  comments: [],
+  loaded: true,
```

Comment:
The initial state for 'loaded' should be false. Only after successful fetching of comments, it should be updated to true.